### ISSUE TITLE
ERROR Found in .fastTSSCounts for creating arrowfile

### ISSUE THREAD

**Issue by macrophage666** [GreenleafLab/ArchR#2127]
Hi,

I have trouble creating an arrowfile. I have non-standard chromosomes for our fragment files, so I customized the geneAnnotation and genomeAnnotation which have the same seqnames with our fragment files but I still got this error. Here is log:
[ArchR-createArrows-2d96605e219c6f-Date-2024-03-04_Time-14-05-54.045259.log](https://github.com/GreenleafLab/ArchR/files/14487180/ArchR-createArrows-2d96605e219c6f-Date-2024-03-04_Time-14-05-54.045259.log)

Thanks

**Comment by rcorces** [comment]
Hi @macrophage666! Thanks for using ArchR! Lately, it has been very challenging for me to keep up with maintenance of this package and all of my other 
responsibilities as a PI. I have not been responding to issue posts and I have not been pushing updates to the software. We are actively searching to hire
a computational biologist to continue to develop and maintain ArchR and related tools. If you know someone who might be a good fit, please let us know!
In the meantime, your issue will likely go without a reply. Most issues with ArchR right not relate to compatibility. Try reverting to R 4.1 and Bioconductor 3.15.
Newer versions of Seurat and Matrix also are causing issues. Sorry for not being able to provide active support for this package at this time.

**Comment by YaoyJiang** [comment]
have you solved it? I met the same error

**Comment by macrophage666** [comment]
> have you solved it? I met the same error
Yes, I solved it. Did you also use a non-standard chromosome? If so, this error could be caused by that. The best way to solve it is to customize your BSgenome and Txdb packages first, using the same fa and gtf files you used to get your fragment files. Then, when creating genome annotation, exclude the unknown chromosomes. Remember to keep chrPrefix as false before creating arrowfiles. 

Here are some tutorials for helping you forge your customized BSgenome and Txdb package:
https://bioconductor.org/packages/release/bioc/html/BSgenomeForge.html
https://bioconductor.org/packages/release/bioc/vignettes/GenomicFeatures/inst/doc/GenomicFeatures.html

It might take some time to get those two packages ready, but there is a high chance that your problems will be solved by doing so. I would like to help if you meet other problems in this process.

**Comment by YaoyJiang** [comment]
> > have you solved it? I met the same error
> > Yes, I solved it. Did you also use a non-standard chromosome? If so, this error could be caused by that. The best way to solve it is to customize your BSgenome and Txdb packages first, using the same fa and gtf files you used to get your fragment files. Then, when creating genome annotation, exclude the unknown chromosomes. Remember to keep chrPrefix as false before creating arrowfiles.
> 
> Here are some tutorials for helping you forge your customized BSgenome and Txdb package: https://bioconductor.org/packages/release/bioc/html/BSgenomeForge.html https://bioconductor.org/packages/release/bioc/vignettes/GenomicFeatures/inst/doc/GenomicFeatures.html
> 
> It might take some time to get those two packages ready, but there is a high chance that your problems will be solved by doing so. I would like to help if you meet other problems in this process.

Thank you very much. I have built the BSgenome and Txdb packages, but I'm still encountering the same error. What confuses me is what you mentioned about excluding the unknown chromosomes. What do you mean by "unknown chromosomes"?

**Comment by YaoyJiang** [comment]
> > have you solved it? I met the same error
> > Yes, I solved it. Did you also use a non-standard chromosome? If so, this error could be caused by that. The best way to solve it is to customize your BSgenome and Txdb packages first, using the same fa and gtf files you used to get your fragment files. Then, when creating genome annotation, exclude the unknown chromosomes. Remember to keep chrPrefix as false before creating arrowfiles.
> 
> Here are some tutorials for helping you forge your customized BSgenome and Txdb package: https://bioconductor.org/packages/release/bioc/html/BSgenomeForge.html https://bioconductor.org/packages/release/bioc/vignettes/GenomicFeatures/inst/doc/GenomicFeatures.html
> 
> It might take some time to get those two packages ready, but there is a high chance that your problems will be solved by doing so. I would like to help if you meet other problems in this process.

And I have posted an issue at [https://github.com/GreenleafLab/ArchR/issues/2149#issue-2246455525](url)

**Comment by macrophage666** [comment]
> > > Have you solved it? I met the same error
> > > Yes, I solved it. Did you also use a non-standard chromosome? If so, this error could be caused by that. The best way to solve it is to customize your BSgenome and Txdb packages first, using the same fa and gtf files you used to get your fragment files. Then, when creating genome annotation, exclude the unknown chromosomes. Remember to keep chrPrefix as false before creating arrowfiles.
> > 
> > 
> > Here are some tutorials for helping you forge your customized BSgenome and Txdb package: https://bioconductor.org/packages/release/bioc/html/BSgenomeForge.html https://bioconductor.org/packages/release/bioc/vignettes/GenomicFeatures/inst/doc/GenomicFeatures.html
> > It might take some time to get those two packages ready, but there is a high chance that your problems will be solved by doing so. I would like to help if you meet other problems in this process.
> 
> And I have posted an issue at [https://github.com/GreenleafLab/ArchR/issues/2149#issue-2246455525](url)

Thanks for providing your original post. Your case is more complicated. I found that the genome for the organism you used only had one assembled chromosome: the chr for mitochondria. The others are just the scaffolds.  In my case, only a few genes were at the scaffold level, so I just excluded those scaffolds in the genome annotation, and it works. You could not do that since most of your genes are in scaffold level. You can try just keep that assembled chromosome for creating arrowfile which probably will work. This post discussed why the scaffold might not work: https://github.com/GreenleafLab/ArchR/issues/88#top
